vllm/v1/sample/logits_processor/__init__.py

```diff
@@ -177,9 +181,125 @@ def build_logitsprocs(
             BUILTIN_LOGITS_PROCESSORS, custom_logitsprocs_classes))
 
 
+V0WrapperState = tuple[LogitsProcessorV0, list[int], list[int]]
+
+
+class LogitsProcessorV0Wrapper(LogitsProcessor):
+    """Wrapper for vLLM V0-style `Callable` logits processors
+    
+    To wrap a specific `Callable` logits processor, 
+    * Implement `self.is_argmax_invariant()` base-class method
+    * Implement `self._new_logits_processor()`
+    * Implement `self._nargs` property
+    
+    """
+
+    def __init__(self, vllm_config: VllmConfig, device: torch.device,
+                 is_pin_memory: bool):
+        # For each request that is assigned a key in self.req_info, store an
+        # optional V0-style logits processor, prompt token ids, and previously
+        # decoded token ids
+        self.req_info: dict[int, V0WrapperState] = {}
+
+        # Preserve config info for use when constructing V0-style logits
+        # processors
+        self.vllm_config = vllm_config
+        self.device = device
+        self.is_pin_memory = is_pin_memory
+
+        # V0 logits processors can have 2 or 3 arguments
+        assert (nargs := self._nargs) in {2, 3}
+        self.nargs = nargs
+
+    @property
+    @abstractmethod
+    def _nargs(self) -> int:
+        """Returns the number of arguments supported by this V0 logits
+        processor. Must be 2 or 3"""
+        raise NotImplementedError()
+
+    @abstractmethod
+    def _new_logits_processor(
```

Comment:
I think better not to use `_` ... it's private in a sense but also part of the public interface to be implemented.

And would be nice to include "request" in the name somehow... perhaps `logits_processor_for_request` though that is a bit long